### PR TITLE
Fix 409 errors for odh-notebook-controller

### PR DESCRIPTION
**Pull request by VaishnaviHire**
During scale tessting, 409 errors were observed for odh-notebook-controller. Possible cause for this is error handling during creation of `trusted-ca` configmap.

## Description
This PR updates the controller to add a `Get` request for the configmap, before creating it.
## Merge criteria:

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Comment by VaishnaviHire** [comment]
Hi folks @harshad16 @LaVLaS , can I get a quick review on this. Its a small change on how we handle errors while creating `trusted-ca` Configmap

File: components/odh-notebook-controller/controllers/notebook_controller.go [hunk 1]
```diff
@@ -200,21 +200,34 @@ func (r *OpenshiftNotebookReconciler) Reconcile(ctx context.Context, req ctrl.Re
...
+	err := r.Client.Get(ctx, client.ObjectKey{
+		Namespace: desiredTrustedCAConfigMap.Namespace,
+		Name:      desiredTrustedCAConfigMap.Name,
+	}, foundTrustedCAConfigMap)
 	if err != nil {
-		if apierrs.IsAlreadyExists(err) {
-			return nil
+		if apierrs.IsNotFound(err) {
+			r.Log.Info("Creating trusted-ca configmap")
+			err = r.Create(ctx, desiredTrustedCAConfigMap)
```
Comment (harshad16) [inline components/odh-notebook-controller/controllers/notebook_controller.go:219]: New this code block
should this be :
```suggestion
			err = r.Client.Create(ctx, desiredTrustedCAConfigMap)
```

**Review by harshad16 — COMMENTED** [review]
lgtm
just one question

File: components/odh-notebook-controller/controllers/notebook_controller.go (hunk 1 above)
Comment (VaishnaviHire) [reply components/odh-notebook-controller/controllers/notebook_controller.go:219]: Fixed

**Review by harshad16 — APPROVED** [review]
/lgtm

:+1:

**Pull request merged**